src/lib/storage/chunk.cpp

```diff
@@ -1,31 +1,51 @@
 #include "chunk.hpp"
 
-#include "abstract_segment.hpp"
 #include "utils/assert.hpp"
 
 namespace opossum {
 
 void Chunk::add_segment(const std::shared_ptr<AbstractSegment> segment) {
-  // Implementation goes here
+  _columns.push_back(segment);
+}
+
+template <typename T>
+bool append_with_type(std::vector<std::shared_ptr<AbstractSegment>>::iterator column_it,
+                      std::vector<AllTypeVariant>::const_iterator value_it) {
+  auto segment = std::dynamic_pointer_cast<ValueSegment<T>>(*column_it);
+  if (segment) {
+    segment->append(*value_it);
+    return true;
+  }
+  return false;
 }
 
 void Chunk::append(const std::vector<AllTypeVariant>& values) {
-  // Implementation goes here
+  DebugAssert(values.size() == column_count(), "Cannot insert a tuple with less values than columns.");
+
+  auto column_it = _columns.begin();
+  auto value_it = values.begin();
+
+  for (; column_it < _columns.end(); ++column_it) {
+    (append_with_type<int32_t>(column_it, value_it) || append_with_type<int64_t>(column_it, value_it) ||
```

Comment:
It would be nice not to duplicate the knowledge of which types are possible for the segment

Reply:
Yes, do you have a suggestion how to do that properly?

Reply:
Not really we also struggled with that, see https://github.com/cambryx/DYOD_SoSe23/blob/sprint1/src/lib/storage/chunk.cpp, they managed to do it